### PR TITLE
Fix random failure of SorterTest.testSorterContentAdd()

### LINKED ISSUE #794: Random failing of SorterTest.testSorterContentAdd()
The test failed during a [custom build](https://github.com/fedejeanne/eclipse.platform.ui/actions/runs/5155048913/jobs/9284234729#step:7:17176) in my forked repository. It was during the freeze period for 4.29 and the code was current, so I'm creating an issue.

Here's the stack trace:

```
org.eclipse.ui.tests.navigator.SorterTest.testSorterContentAdd  Time elapsed: 0.365 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<Blue[Added]Parent> but was:<Blue[]Parent>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.eclipse.ui.tests.navigator.SorterTest.testSorterContentAdd(SorterTest.java:291)
```

### PR DESCRIPTION
**Pull request by fedejeanne**
The test relies on `TestLabelProviderBlue`, which is not being instantly initialized. This behavior seems to be known and accepted since the test `LabelProviderTest.testChangeActivation()` was already using a busy wait.

This PR extracts a new method into `NavigatorTestBase` and uses it in both tests. The method also asserts that the required label provider has been initialized, which helps debugging.

fixes #794

### CODE REVIEW

**Review by BeckerWdf — APPROVED** [review]

**Comment by fedejeanne** [comment]
@akurtakov thank you for merging :)

I've a question though: if I close this PR, I see the following message: "Closed with unmerged commits." . 

**Most importantly: I also see that the PR has a red icon instead of the violet icon, which makes it look like the PR was dropped instead of merged.**

Should I do something differently in order to close this PR as "merged" and show the violet icon instead of the red one?

Thank you in advance

**Comment by akurtakov** [comment]
I haven't merged this one. I rebased it on top of master for the automation to kick in.

**Comment by fedejeanne** [comment]
I bumped the version (+100) for `org.eclipse.ui.tests.navigator`. This PR now needs approval from a maintainer for the automation to kick in.

**Comment by iloveeclipse** [comment]
@fedejeanne : please ping me once tests complete.

**Pull request merged**
